### PR TITLE
Allow slash.use_fixtures on other fixtures (issue 1013)

### PR DESCRIPTION
**Pull request by yaelmi3**
https://github.com/getslash/slash/issues/1013

### CODE REVIEW

File: tests/test_fixture_use.py [hunk 1]
```diff
@@ -50,6 +50,31 @@ def test_a():  # pylint: disable=unused-variable
...
+    @suite_builder.first_file.add_code
+    def __code__():  # pylint: disable=unused-variable
+        import slash
+
+        @slash.fixture
+        def fixture_1():
+            slash.context.result.data['params'] = ["fixture_1"]
+
+        @slash.fixture
+        @slash.use_fixtures(["fixture_3"])
```
Comment (vmalloc) [inline tests/test_fixture_use.py:63]: I'd use another fixture here where the order of the decorators is reversed. In the past several bugs manifested themselves this way

File: slash/core/fixtures/fixture_store.py [hunk 2]
```diff
@@ -255,25 +255,35 @@ def get_fixture_by_id(self, fixture_id):
...
             if required_name in skip_names:
                 continue
             if element == 'this':
                 continue
             fixture = namespace.get_fixture_by_name(real_name)
-            if get_values:
-                fixture = self.get_fixture_value(fixture, name=required_name)
-            returned[required_name] = fixture
-        return returned
+            if (required_name, fixture) not in fixtures_list:
```
Comment (vmalloc) [inline slash/core/fixtures/fixture_store.py:281]: I think fixture_list should actually be a set. I'm not 100% sure if this has implications, but this nested lookup can be slow in extreme cases...

File: slash/core/fixtures/fixture_store.py [hunk 3]
```diff
@@ -62,8 +62,8 @@ def call_with_fixtures(self, test_func, namespace, trigger_test_start=False, tri
             used_fixtures_decorator_names = getattr(test_func, '__extrafixtures__', None)
```
Comment (vmalloc) [inline slash/core/fixtures/fixture_store.py:62]: This change requires a changelog entry, IMO

**Comment by vmalloc** [comment]
Seems like the CI is failing on this?

**Comment by yaelmi3** [comment]
@vmalloc : the CI seems to fails on tests/test_parallel.py, a timeout issue? Ran it locally on my end, looks ok. `ERROR: slash.parallel.parallel_manager: No request sent to server for 2 seconds, terminating`. Not familiar with Travis, is there an option run it again?

**Comment by ayalash** [comment]
@yaelmi3 Yeah, Travis is sometimes acting out...
I rerun the job

**Comment by ayalash** [comment]
@vmalloc LGTM

**Comment by vmalloc** [comment]
Thanks a lot @yaelmi3 !

**Pull request merged**